### ISSUE TITLE
p7zip 15.14.1 fails to build on Darwin

### ISSUE THREAD

**Issue by clacke** [NixOS/nixpkgs#15152]
## Issue description

The new p7zip version has a new structure for its makefiles, which doesn't fit the Nix build definition on the OSX side.

### Steps to reproduce

```
$ nix-env -iA nixpkgs.p7zip
replacing old ‘p7zip-9.38’
installing ‘p7zip-15.14.1’
download-from-binary-cache.pl: could not download ‘https://cache.nixos.org/n4hr88vc8pqx5id87azljv7x8mnz9gjd.narinfo’ (Curl error 77)
these derivations will be built:
  /nix/store/mskrzqrbk33zspw82vgc1x2pa83faycr-p7zip-15.14.1.drv
download-from-binary-cache.pl: could not download ‘https://cache.nixos.org/n4hr88vc8pqx5id87azljv7x8mnz9gjd.narinfo’ (Curl error 77)
building path(s) ‘/nix/store/n4hr88vc8pqx5id87azljv7x8mnz9gjd-p7zip-15.14.1’
unpacking sources
unpacking source archive /nix/store/6bdyks74fc7kb9zajrzxzqzssw6y0sh8-p7zip_15.14.1_src_all.tar.bz2
source root is p7zip_15.14.1
setting SOURCE_DATE_EPOCH to timestamp 1458761867 of file p7zip_15.14.1/makefile.machine
patching sources
configuring
cp: cannot stat 'makefile.macosx_64bits': No such file or directory
builder for ‘/nix/store/mskrzqrbk33zspw82vgc1x2pa83faycr-p7zip-15.14.1.drv’ failed with exit code 1
error: build of ‘/nix/store/mskrzqrbk33zspw82vgc1x2pa83faycr-p7zip-15.14.1.drv’ failed

$ tar atf /nix/store/6bdyks74fc7kb9zajrzxzqzssw6y0sh8-p7zip_15.14.1_src_all.tar.bz2 | grep makefile.*osx
p7zip_15.14.1/makefile.macosx_llvm_64bits
p7zip_15.14.1/makefile.macosx_gcc_64bits
p7zip_15.14.1/makefile.macosx_gcc_32bits
p7zip_15.14.1/makefile.macosx_gcc_32bits_ppc
p7zip_15.14.1/makefile.macosx_gcc_32bits_asm
```

## Technical details

* System: OSX: Darwin Kernel Version 15.4.0 (El Capitan 10.11.4)
* Nix version: 1.11.2
* Nixpkgs version: 16.09pre82421.d27848a

**Comment by ChrisJefferson** [comment]
Fixed by #15715 (hopefully!)

**Comment by clacke** [comment]
http://hydra.nixos.org/build/36245924 worked.

Now it is failing for unrelated reasons. Filing a new issue for that.
